src/main/java/seedu/exercise/logic/parser/ArgumentTokenizer.java

```diff
@@ -21,24 +22,38 @@
      *
      * @param argsString Arguments string of the form: {@code preamble <prefix>value <prefix>value ...}
      * @param prefixes   Prefixes to tokenize the arguments string with
-     * @return           ArgumentMultimap object that maps prefixes to their arguments
+     * @return ArgumentMultimap object that maps prefixes to their arguments
      */
     public static ArgumentMultimap tokenize(String argsString, Prefix... prefixes) {
         List<PrefixPosition> positions = findAllPrefixPositions(argsString, prefixes);
         return extractArguments(argsString, positions);
     }
 
+    /**
+     * Tokenizes an arguments string and returns an {@code ArgumentMultimap} object that maps prefixes to their
+     * respective argument values. Only the given prefixes will be recognized in the arguments string.
+     *
+     * @param argsString Arguments string of the from: {@code preamble <prefix>value <prefix>value ...}
+     * @param prefixes   Prefixes to tokenize the arguments string with
+     * @return ArgumentMultimap object that maps prefixes to their arguments
+     */
+    public static ArgumentMultimap tokenize(String argsString, Set<Prefix> prefixes) {
+        int numOfPrefixes = prefixes.size();
```

Comment:
Might want to explain why this method has been overloaded. Does it replace the above completely such that `findAllPrefixPositions(argsString, prefixes)` and `extractArguments(argsString, positions)` are  no longer needed? Or does it serve a different use case?

Is it possible to solely rely on the original one above? Perhaps, you may consider abstracting out the handling of custom properties prefixes into a separate method, and then use this method in the original tokenize.

Reply:
Removed the additional method and included a new method in ``CliSyntax`` that will help to combine the additional prefixes required for ``Add`` and ``Edit`` commands with the property prefixes.

Also removed the prefixes ``PREFIX_INDEX`` and ``PREFIX_CATEGORY`` from the default property prefixes as they are not prefixes for exercise's properties.